authenticate/handlers.go

```diff
@@ -273,20 +273,23 @@ func (a *Authenticate) signOutRedirect(w http.ResponseWriter, r *http.Request) e
 	}
 
 	endSessionURL, err := authenticator.LogOut()
-	if err == nil && redirectString != "" {
+	if err != nil {
+		if !errors.Is(err, oidc.ErrSignoutNotImplemented) {
+			log.Warn(r.Context()).Err(err).Msg("authenticator.LogOut")
+		}
+		return httputil.NewError(http.StatusOK, errors.New("user logged out"))
+	}
+
+	if redirectString != "" {
 		params := url.Values{}
 		params.Add("id_token_hint", rawIDToken)
 		params.Add("post_logout_redirect_uri", redirectString)
 		endSessionURL.RawQuery = params.Encode()
-		redirectString = endSessionURL.String()
-	} else if err != nil && !errors.Is(err, oidc.ErrSignoutNotImplemented) {
-		log.Warn(r.Context()).Err(err).Msg("authenticate.SignOut: failed getting session")
 	}
-	if redirectString != "" {
-		httputil.Redirect(w, r, redirectString, http.StatusFound)
-		return nil
-	}
-	return httputil.NewError(http.StatusOK, errors.New("user logged out"))
+	redirectString = endSessionURL.String()
+
+	httputil.Redirect(w, r, redirectString, http.StatusFound)
+	return nil
```

Comment:
I think the logic here has changed. I don't know what these subtle changes in behavior will lead to.

I think we won't redirect anymore if sign out is not implemented?